### PR TITLE
SONARJAVA-794 SuppressWarnings in Trivia

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [SuppressWarningFilter.java#L35](https://github.com/SonarSource/sonar-java/blob/60893a5faf1c6be4b3eb0be2126f3834cfc81db0/java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java#L35): Remove this unused import 'org.sonar.java.model.JavaTree'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ private void handleSuppressWarning(List<AnnotationTree> annotationTrees, Tree tr
     List<String> rules = Lists.newArrayList();
     for (AnnotationTree annotationTree : annotationTrees) {
       if (isSuppressWarningsAnnotation(annotationTree)) {
-        startLine = ((JavaTree) annotationTree).getLine();
+        startLine = startLineIncludingTrivia(tree);
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: The start line computation was starting from the annotation, now you are starting from the tree, which is not the same.
Moreover, I have the feeling that it won't work if there is comment in-between annotations (see code below), probably worth to add the case.

```
class A {

  /**
   * TODO
   */
  @MyAnnotation
  /**
   * TODO
   */
  @SuppressWarnings("repo:S1135")
  void foo() {

  }
}

@interface MyAnnotation {
}
```

Comment (saberduck) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: Yes, the start line computation has to be modified to account for the javadoc, why is that an issue? The example you gave works as it should. Perhaps I didn't understood the comment? What kind of behavior you expect?

Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: Indeed, I miss-read the code. Sorry!
To me we were only excluding the lines from the trivia attached to that the very first token, and not selecting the highest line as start-line for the whole exclusion.

**Pull request merged**